ui/hooks/src/use-beams.ts

```diff
@@ -237,21 +246,43 @@ export const useBeams = ({ overscan, filters, sorting, did }: UseBeamsOptions) =
 
   const fetchInitialData = React.useCallback(
     async (restoreItem?: { key: string; offsetTop: number }) => {
-      if (beamsQuery.called) return;
+      if ((beamsQuery.called && !(!isLoggedIn && !authenticating && showNsfw)) || authenticating)
+        return;
 
       const initialVars: GetBeamStreamQueryVariables & GetBeamsByAuthorDidQueryVariables = {
+        ...mergedVars,
         sorting: { createdAt: SortOrder.Desc },
         id: did ?? undefined,
         indexer: did ? undefined : sdk.services.gql.indexingDID,
       };
 
+      if (!showNsfw || !isLoggedIn) {
+        initialVars.filters = {
+          and: [{ where: { status: { isNull: true } } }],
+        };
+      }
+
+      if (showNsfw && isLoggedIn) {
+        initialVars.filters = null;
```

Comment:
this does not follow what we had discussed last time, it should have been an `or` filter; what has changed since our discussion?

Reply:
Yes but the problem is that there are multiple status levels(ex: suspended) which would be loaded if I check nsfw.